### PR TITLE
TASK-2257 | Core navigate returns promise

### PR DESCRIPTION
**Pull request by rafalgamon**
**Description**

Changes proposed in this pull request:

- Core navigate returns promise

### CODE REVIEW

File: core/src/utilities/helpers/generic-helpers.js [hunk 1]
```diff
@@ -209,26 +209,23 @@ class GenericHelpersClass {
     return processedString;
   }
 
-  getInnerHeight() /* istanbul ignore next */ {
+  getInnerHeight() {
     return LuigiElements.isCustomLuigiContainer() ? LuigiElements.getLuigiContainer().clientHeight : window.innerHeight;
   }
 
-  getContentAreaHeight /* istanbul ignore next */() {
```
Comment (JohannesDoberer) [inline core/src/utilities/helpers/generic-helpers.js:216]: Why did you remove the comments?

File: test/e2e-test-application/e2e/tests/1-angular/i18n-context-product-app-switcher.spec.js [hunk 2]
```diff
@@ -106,6 +106,7 @@ describe('Context switcher', () => {
 
   it('Preserve sub path on switch', () => {
     cy.window().then(win => {
+      cy.wait(300);
```
Comment (JohannesDoberer) [inline test/e2e-test-application/e2e/tests/1-angular/i18n-context-product-app-switcher.spec.js:109]: Locally I don't need the all waitings. Is it necessary?

**Review by JohannesDoberer — CHANGES_REQUESTED** [review]

File: core/src/utilities/helpers/generic-helpers.js [hunk 3]
```diff
@@ -316,6 +313,44 @@ class GenericHelpersClass {
     }
     return val;
   }
+
+  createRemotePromise() {
```
Comment (JohannesDoberer) [inline core/src/utilities/helpers/generic-helpers.js:317]: Unit tests would be nice

File: core/src/utilities/helpers/generic-helpers.js [hunk 4]
```diff
@@ -316,6 +313,44 @@ class GenericHelpersClass {
...
+    let res, rej;
+    const prom = new Promise((resolve, reject) => {
+      res = () => {
+        resolve();
+      };
+      rej = () => {
+        reject(new Error());
```
Comment (JohannesDoberer) [inline core/src/utilities/helpers/generic-helpers.js:324]: When I try to open a route twice I'm getting an error in the console
Bildschirmfoto 2022-02-10 um 08 57 49

File: core/src/core-api/_internalLinkManager.js [hunk 5]
```diff
@@ -43,10 +45,12 @@ export class linkManager extends LuigiCoreAPIBase {
...
-      })
+      }),
+      remotePromiseId: remotePromise.id
     };
 
     this.sendPostMessageToLuigiCore(navigationOpenMsg);
+    return remotePromise;
   }
 
   openAsModal(path, modalSettings = {}) {
```
Comment (JohannesDoberer) [inline core/src/core-api/_internalLinkManager.js:56]: I would assume open as modal, drawer, splitview should also return a promise.
If it is too much work also fine in follow-up task. If so could you create this task?

File: core/src/utilities/helpers/generic-helpers.js (hunk 1 above)
Comment (rafalgamon) [reply core/src/utilities/helpers/generic-helpers.js:216]: I think i was something from auto format file but I revers all these changes

File: core/src/utilities/helpers/generic-helpers.js (hunk 3 above)
Comment (rafalgamon) [reply core/src/utilities/helpers/generic-helpers.js:317]: done

File: core/src/core-api/_internalLinkManager.js (hunk 5 above)
Comment (rafalgamon) [reply core/src/core-api/_internalLinkManager.js:56]: I will create new Task for this and implement it in this task.

File: core/src/utilities/helpers/generic-helpers.js (hunk 1 above)
Comment (JohannesDoberer) [reply core/src/utilities/helpers/generic-helpers.js]: It belongs to our code coverage.

File: core/src/utilities/helpers/generic-helpers.js (hunk 4 above)
Comment (JohannesDoberer) [reply core/src/utilities/helpers/generic-helpers.js:326]: If unsaved changes warning modal is displayed and dismissed I'll get the error too.
You can reproduce it when you go to _overview_ in our e2e test app and turn on the toggle _There are unsaved changes_. Try to navigate somewhere else like in the attached screenshot and click on _somewhere else_ and dismiss the warnings modal.

Bildschirmfoto 2022-02-14 um 16 21 26

Comment (rafalgamon) [reply core/src/utilities/helpers/generic-helpers.js:326]: Unfortunately I can reproduce it. I merged and resolved all conflict between this branch and `master`  so maybe now there is't this error.

File: docs/luigi-client-api.md [hunk 6]
```diff
@@ -507,6 +507,9 @@ Navigates to the given path in the application hosted by Luigi. It contains eith
     -   `drawerSettings.backdrop` **[boolean](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/Boolean)** By default, it is set to `false`. If it is set to `true` the rest of the screen has a backdrop.
     -   `drawerSettings.size` **(`"l"` \| `"m"` \| `"s"` \| `"xs"`)** size of the drawer (optional, default `"s"`)
 
+##### Return
```
Comment (alexandra-simeonova) [inline docs/luigi-client-api.md:510]: I'm not sure what needs to be documented here. Is "return" a new API function? If so, please update the API .js file with the documentation as well.

Comment (JohannesDoberer) [reply docs/luigi-client-api.md:510]: Removed it from luigi client api

**Review by alexandra-simeonova — APPROVED** [review]

**Review by JohannesDoberer — APPROVED** [review]
